lib/request.js

```diff
@@ -50,7 +50,7 @@ function getTrustProxyFn (tp) {
   }
   if (tp === true) {
     // Support plain true/false
-    return function () { return true }
+    return undefined
```

Comment:
Nit: this is better than explicit undefined to represent the absence of a function/object imo

```suggestion
    return null
```

Reply:
I agree